.github/workflows/build-run-app.yml

```diff
@@ -45,9 +45,11 @@ jobs:
     steps:
       - name: Set up environment variables
         # Apps typically use lowercase targets and uppercase names, this logic is fragile but works
+        # Mark the sample app/lib and lab apps as we don't need to add them to the build or startup file (they're already part of the bundle)
         run: |
           echo "APP_UPPER=$(echo ${{ inputs.app-name }} | sed 's/[a-z]/\U&/g')" >> $GITHUB_ENV
           echo "APP_LOWER=$(echo ${{ inputs.app-name }} | sed 's/[A-Z]/\L&/g')" >> $GITHUB_ENV
+          echo "IS_SAMPLE_OR_LAB=$(echo ${{ inputs.app-name }} | grep -iqE 'sample|lab' && echo true || echo false)" >> $GITHUB_ENV
```

Comment:
My only (slight) concern is with this logic.  I can see why you did this -- that the "sample" and "lab" apps are already in the sample config, thus we need to _not_ add them again or else we'd end up with a duplicate entry.

My concern is that this will break if someone changes the sample config such that this assumption is not true anymore.  I also don't want to assume that there will never be another app with either "sample" or "lab" in its name.

Could we do something more direct -- that is, instead of checking the app name, check if the script already has the entry?  That is, if running on "to_lab" -- just check if the `cfe_es_startup.scr` already has an entry for to_lab, and skip the update if it does?

Reply:
At the very least -- another idea is could this be a parameter that is passed into the workflow?  Something like "is_framework_app" ... with a default value of false.  Then when TO/CI/SCH Lab and Sample App invoke this workflow, they can pass it as true, but everything else remains as it was.  This would accomplish the same goal without making this assumption.